src/client/pkg/grpcutil/buffer.go

```diff
@@ -4,24 +4,45 @@ import (
 	"sync"
 )
 
-var bufPool = sync.Pool{
-	New: func() interface{} {
-		// This buffer size is:
-		// 1. Reasonably smaller than the max gRPC size
-		// 2. Small enough that having hundreds of these buffers won't
-		// overwhelm the node
-		// 3. Large enough for message-sending to be efficient
-		return make([]byte, MaxMsgSize/10)
-	},
+// BufPool is a wrapper around sync.Pool that makes it a little nicer to use
+// for []byte by doing the casting for you and defining the `New` function.
+type BufPool struct {
+	sync.Pool
+}
+
+// NewBufPool creates a new BufPool that returns buffers of the given size.
+func NewBufPool(size int) *BufPool {
+	return &BufPool{sync.Pool{
+		New: func() interface{} { return make([]byte, size) },
+	}}
 }
 
 // GetBuffer returns a buffer.  The buffer may or may not be freshly
 // allocated, and it may or may not be zero-ed.
-func GetBuffer() []byte {
+func (b *BufPool) GetBuffer() []byte {
 	return bufPool.Get().([]byte)
```

Comment:
This should be `return b.Get().([]byte)`

Reply:
Yeah... very good point.